dissect/protos/dns.py

```diff
@@ -0,0 +1,575 @@
+
+try:
+     xrange = xrange
+except:
+     xrange = range
+
+from vstruct2.types import *
+import dissect.protos.inet as vs_inet
+
+# only the currently parsed record types
+DNS_TYPE_A     = 1
+DNS_TYPE_NS    = 2
+DNS_TYPE_CNAME = 5
+DNS_TYPE_SOA   = 6
+DNS_TYPE_PTR   = 12
+DNS_TYPE_MX    = 15
+DNS_TYPE_TXT   = 16
+DNS_TYPE_RP    = 17
+DNS_TYPE_AFSDB = 18
+DNS_TYPE_SIG   = 24
+DNS_TYPE_KEY   = 25
+DNS_TYPE_AAAA  = 28
+DNS_TYPE_LOC   = 29
+DNS_TYPE_SRV   = 33
+DNS_TYPE_NAPTR = 35
+DNS_TYPE_KX    = 36
+DNS_TYPE_CERT  = 37
+DNS_TYPE_DNAME = 39
+
+DNS_TYPE_OPT   = 41
+DNS_TYPE_APL   = 42
+DNS_TYPE_DS    = 43
+DNS_TYPE_SSHFP = 44
+DNS_TYPE_IPSECKEY = 45
+DNS_TYPE_RRSIG = 46
+DNS_TYPE_NSEC  = 47
+DNS_TYPE_DNSKEY= 48
+DNS_TYPE_DHCID = 49
+
+DNS_TYPE_NSEC3 = 50
+DNS_TYPE_NSEC3PARAM = 51
+DNS_TYPE_TLSA  = 52
+DNS_TYPE_HIP   = 55
+DNS_TYPE_CDS   = 59
+DNS_TYPE_CDNSKEY = 60
+
+DNS_TYPE_TKEY  = 249
+DNS_TYPE_TSIG  = 250
+DNS_TYPE_IXFR  = 251
+DNS_TYPE_AXFR  = 252
+DNS_TYPE_ANY   = 255
+DNS_TYPE_URI   = 256
+DNS_TYPE_CAA   = 257
+
+DNS_TYPE_TA    = 32768
+DNS_TYPE_DLV   = 32769
+
+dns_type_names = {
+    DNS_TYPE_A:     'A',
+    DNS_TYPE_NS:    'NS',
+    DNS_TYPE_CNAME: 'CNAME',
+    DNS_TYPE_SOA:   'SOA',
+    DNS_TYPE_PTR:   'PTR',
+    DNS_TYPE_MX:    'MX',
+    DNS_TYPE_TXT:   'TXT',
+    DNS_TYPE_RP    : 'RP',
+    DNS_TYPE_AFSDB : 'AFSDB',
+    DNS_TYPE_SIG   : 'SIG',
+    DNS_TYPE_KEY   : 'KEY',
+    DNS_TYPE_AAAA  : 'AAAA',
+    DNS_TYPE_LOC   : 'LOC',
+    DNS_TYPE_SRV   : 'SRV',
+    DNS_TYPE_NAPTR : 'NAPTR',
+    DNS_TYPE_KX    : 'KX',
+    DNS_TYPE_CERT  : 'CERT',
+    DNS_TYPE_DNAME : 'DNAME',
+    DNS_TYPE_OPT   : 'OPT',
+    DNS_TYPE_APL   : 'APL',
+    DNS_TYPE_DS    : 'DS',
+    DNS_TYPE_SSHFP : 'SSHFP',
+    DNS_TYPE_IPSECKEY : 'IPSECKEY',
+    DNS_TYPE_RRSIG : 'RRSIG',
+    DNS_TYPE_NSEC  : 'NSEC',
+    DNS_TYPE_DNSKEY: 'DNSKEY',
+    DNS_TYPE_DHCID : 'DHCID',
+    DNS_TYPE_NSEC3 : 'NSEC3',
+    DNS_TYPE_NSEC3PARAM : 'NSEC3PARAM',
+    DNS_TYPE_TLSA  : 'TLSA',
+    DNS_TYPE_HIP   : 'HIP',
+    DNS_TYPE_CDS   : 'CDS',
+    DNS_TYPE_CDNSKEY : 'CDNSKEY',
+    DNS_TYPE_TKEY  : 'TKEY',
+    DNS_TYPE_TSIG  : 'TSIG',
+    DNS_TYPE_IXFR  : 'IXFR',
+    DNS_TYPE_AXFR  : 'AXFR',
+    DNS_TYPE_ANY   : 'ANY',
+    DNS_TYPE_URI   : 'URI',
+    DNS_TYPE_CAA   : 'CAA',
+    DNS_TYPE_TA    : 'TA',
+    DNS_TYPE_DLV   : 'DLV',
+}
+
+DNS_CLASS_IN     = 1
+DNS_CLASS_CSNET  = 2
+DNS_CLASS_CHAOS  = 3
+DNS_CLASS_HESIOD = 4
+DNS_CLASS_ANY    = 255
+
+dns_class_names = {
+    DNS_CLASS_IN     : 'IN',
+    DNS_CLASS_CSNET  : 'CSNET',
+    DNS_CLASS_CHAOS  : 'CHAOS',
+    DNS_CLASS_HESIOD : 'HESIOD',
+    DNS_CLASS_ANY    : 'ANY',
+}
+
+DNS_NAMETYPE_LABEL        = 0
+DNS_NAMETYPE_RESERVED     = 1
+DNS_NAMETYPE_EXTENDED     = 2 
+DNS_NAMETYPE_POINTER      = 3
+DNS_NAMETYPE_LABELPOINTER = 4
+
+DNS_FLAG_RESP   = 1 << 15
+DNS_FLAG_TRUNC  = 1 << 9
+DNS_FLAG_RECUR  = 1 << 8
+DNS_FLAG_AD     = 1 << 5
+#DNS_FLAG_AA = 1 << 5    # authoritative answer
```

Comment:
You know, that's a good q! I am not sure, they were commented out in the original version from vs1, and I didn't see a reason to change them. Tests seem to pass with or without them, but I suspect that's because they are for some case that isn't yet covered by tests.